### DISCUSSION TITLE
Gitlab Authentication failure

### DISCUSSION THREAD

**Discussion by 23ewrdtf** [renovatebot/renovate#17133] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.152.0

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

running on kubernetes to scan our repos on gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I was sent here from https://github.com/renovatebot/helm-charts/issues/243

According to the docs https://docs.renovatebot.com/modules/platform/gitlab/ read only access is supported for dry-runs.

Deployed using process below:

Added personal token (for RENOVATE_TOKEN) to a gitlab user with below permissions (read only for testing):
```
read_api, read_user, read_repository, read_registry
```

Applied below:
```
kubectl apply -f secret.yaml
helm upgrade --namespace renovate renovate renovate/renovate -f values.yaml --debug
```

secret.yaml
```
apiVersion: v1
kind: Secret
metadata:
  name: renovate-env
  namespace: renovate
type: Opaque
stringData:
  RENOVATE_TOKEN: 'xxxxxxx'
```

Values (only changed values shown) (I found how to do it here https://github.com/renovatebot/helm-charts/issues/198#issuecomment-1164201297):
```
renovate:
  existingConfigFile: "/usr/src/app/config.js"

existingSecret: 'renovate-env'

extraConfigmaps:
  - name: config
    data:
      config.js: |
        module.exports = {
          platform: "gitlab",
          autodiscover: "true",
          secrets: {
            RENOVATE_TOKEN: process.env.RENOVATE_TOKEN
          }
        }

extraVolumes:
  - name: renovate-extra-config
    configMap:
      name: renovate-extra-config

extraVolumeMounts:
  - name: renovate-extra-config
    mountPath: /usr/src/app/config.js
    subPath: config.js
```

Error:

```
kubectl logs test-kc5zf -n renovate
 WARN: Config needs migrating
       "originalConfig": {
         "platform": "gitlab",
         "autodiscover": "true",
         "secrets": {"RENOVATE_TOKEN": "***********"}
       },
       "migratedConfig": {
         "platform": "gitlab",
         "autodiscover": true,
         "secrets": {"RENOVATE_TOKEN": "***********"}
       }
FATAL: Authentication failure
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "DCigCzk18VgMnevu9NUXL",
           "msg": "Authentication failure"
         }
       ]
```

Question:
Is the RENOVATE_TOKEN enough to get authenticated to Gitlab? How do I specify user for the RENOVATE_TOKEN?

Do I need to specify anything else apart from RENOVATE_TOKEN to access gitlab repos ?

### Relevant debug logs

<details><summary>Logs</summary>

```
 WARN: Config needs migrating
       "originalConfig": {
         "platform": "gitlab",
         "autodiscover": "true",
         "secrets": {"RENOVATE_TOKEN": "***********"}
       },
       "migratedConfig": {
         "platform": "gitlab",
         "autodiscover": true,
         "secrets": {"RENOVATE_TOKEN": "***********"}
       }
FATAL: Authentication failure
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "DCigCzk18VgMnevu9NUXL",
           "msg": "Authentication failure"
         }
       ]
 ```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by viceice** [comment]
I send you to discussions, as i'm pretty sure this ia  an configuration error.

**Reply by 23ewrdtf** [reply]
Yes, most likely. Sorry, i wasn't aware of the discussion option. Thank you.

**Comment by viceice** [comment, marked as answer]
https://docs.renovatebot.com/modules/platform/gitlab/
☝️ 

readonly token doesn't work with autodiscover, so you need to configure the repos explicit

you should instead use a write token and set renovate `dryRun`[^1]

[^1]: https://docs.renovatebot.com/self-hosted-configuration/#dryrun

**Reply by viceice** [reply]
also set `LOG_LEVEL=debug`[^1] to see more about your issues

[^1]: https://docs.renovatebot.com/examples/self-hosting/#logging
